txscript/engine.go

```diff
@@ -75,6 +75,7 @@ var halfOrder = new(big.Int).Rsh(btcec.S256().N, 1)
 
 // Engine is the virtual machine that executes scripts.
 type Engine struct {
+	sigCache        *SigCache
```

Comment:
While it doesn't really matter from a functionality standpoint, this should probably go after `flags` since it mirrors the order it is specified in `New`.